examples/plot_conditional_vs_marginal_xor_data.py

```diff
@@ -0,0 +1,140 @@
+fig, ax = plt.subplots()
+
+ax.imshow(
+    Z.reshape(xx.shape),
+    interpolation="nearest",
+    extent=(xx.min(), xx.max(), yy.min(), yy.max()),
+    cmap="RdYlBu_r",
+    alpha=0.6,
+    origin="lower",
+    aspect="auto",
+)
+sns.scatterplot(
+    x=X[:, 0],
+    y=X[:, 1],
+    hue=Y,
+    ax=ax,
+    palette="muted",
+)
+ax.axis("off")
+ax.legend(
+    title="Class",
+)
+ax.set_title("Decision function of SVC with RBF kernel")
+plt.show()
+
+
+###############################################################################
+# Computing the conditional and marginal importance
+# -----------------------------------------------------
+# We first compute the marginal importance by fitting univariate models on each feature.
+# Then, we compute the conditional importance using the CPI class. The univarariate
+# models don't perform above chance, since solving the XOR problem requires to use both
+# features. The conditional importance, on the other hand, reveals that both features
+# are important (therefore rejecting the null hypothesis
+# :math:`Y \perp\!\!\!\perp X^1 | X^2`).
+cv = KFold(n_splits=5, shuffle=True, random_state=0)
+clf = SVC(kernel="rbf", random_state=0)
+# Compute marginal importance using univariate models
+marginal_scores = []
+for i in range(X.shape[1]):
+    feat_scores = []
+    for train_index, test_index in cv.split(X_train):
+        X_train_cv, X_test_cv = X_train[train_index], X_train[test_index]
+        y_train_cv, y_test_cv = y_train[train_index], y_train[test_index]
+
+        X_train_univariate = X_train_cv[:, i].reshape(-1, 1)
+        X_test_univariate = X_test_cv[:, i].reshape(-1, 1)
+
+        univariate_model = clone(clf)
+        univariate_model.fit(X_train_univariate, y_train_cv)
+
+        feat_scores.append(univariate_model.score(X_test_univariate, y_test_cv))
+    marginal_scores.append(feat_scores)
+
+
+importances = []
+for i, (train_index, test_index) in enumerate(cv.split(X_train)):
+    X_train_cv, X_test_cv = X_train[train_index], X_train[test_index]
+    y_train_cv, y_test_cv = y_train[train_index], y_train[test_index]
+
+    clf_c = clone(clf)
+    clf_c.fit(X_train_cv, y_train_cv)
+
+    vim = CPI(
+        estimator=clf_c,
+        method="decision_function",
+        loss=hinge_loss,
+        imputation_model_continuous=RidgeCV(np.logspace(-3, 3, 10)),
+        n_permutations=50,
+        random_state=0,
+    )
+    vim.fit(X_train_cv, y_train_cv)
+    importances.append(vim.score(X_test_cv, y_test_cv)["importance"])
+importances = np.array(importances).T
+fig, axes = plt.subplots(1, 2, sharey=True, figsize=(6, 2.5))
+
+# Marginal scores boxplot
+sns.boxplot(
+    data=np.array(marginal_scores).T,
+    orient="h",
+    ax=axes[0],
+    fill=False,
+    color="C0",
+    linewidth=3,
+)
+axes[0].axvline(x=0.5, color="k", linestyle="--", lw=3)
+axes[0].set_ylabel("")
+axes[0].set_yticks([0, 1], ["X1", "X2"])
+axes[0].set_xlabel("Marginal Scores (accuracy)")
```

Comment:
I can't see the ax labels on the generated figure

Reply:
`plt.tight_layout` seems to have solved the issue.